File: design-patterns/python/python-adapter-design-pattern/Mssql.py
class Mssql():
    def __init__(self):
        self.id_counter = 0
        self.user_list = []

    def user_by_id(self, id):
        for user in self.user_list:
            if user.id == id:
                return user
        return None

    def user(self, user):
        self.id_counter += 1
        user.id = self.id_counter
        self.user_list.append(user)

File: design-patterns/csharp/csharp-adapter-design-pattern/Mssql.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace csharp_adapter
{
    public class Mssql
    {
        private readonly List<User> userList;
        private int idCounter;

        public Mssql(){
            idCounter = 0;
            userList = new List<User>();
        } 

        // Methods that do not comply with the DB interface 
        public void user(User user){
            user.Id = idCounter++;
            userList.Add(user);
        }
        public User UserById(int id){
            foreach(User user in userList){
                if(user.Id == id){
                    return user;
                }
            }
            return null;
        }
    }
}